src/lifecycle.ts

```diff
@@ -13,24 +14,11 @@ export type Lifecycles =
   'beforeUpdate' | 'updated' |
   'activated' | 'deactivated'
 
-// Definitions of Location and RawLocation come from source of 'vue-router'
-export interface Location {
-  name?: string;
-  path?: string;
-  hash?: string;
-  query?: Map<string>;
-  params?: Map<string>;
-  append?: boolean;
-  replace?: boolean;
-}
-
-export type RawLocation = string | Location;
-
 export type RouterLifecycle = 'beforeRouteUpdate'|'beforeRouteLeave'
 export type BeforeRouteEnterLifecycle = 'beforeRouteEnter'
 
-export declare type NextFunc = (to?: RawLocation | false | void) => void;
-export declare type NextFuncVm<T extends Vue> = (to?: RawLocation | false | ((vm: T) => any) | void) => void;
+export declare type NextFunc = (to?: RawLocation | Error | false) => void;
+export declare type NextFuncVm<T extends Vue> = (to?: RawLocation | Error | false | ((vm: T) => any)) => void;
```

Comment:
I think generic signature is good enough. We don't have a way to specify `this` in decorator for now.